src/background/vpncontroller/vpncontroller.js

```diff
@@ -218,19 +179,131 @@ export class VPNController extends Component {
    * tcp handle (which is now invalid) is not reused.
    *
    * @readonly
-   * @type {number}
+   * @type {IBindable<Number>}
    */
   get isolationKey() {
     return this.#isolationKey;
   }
 
   #increaseIsolationKey() {
-    ++this.#isolationKey;
+    this.#isolationKey.set(this.#isolationKey.value++);
   }
 
   /** @type {browser.runtime.Port?} */
   #port = null;
-  #isolationKey = 0;
+  #isolationKey = property(0);
+
+  #mState = property(new VPNState());
+  /** @type {WritableProperty<Array<ServerCountry>>} */
+  // @ts-ignore
+  #mServers = property([]);
+
+  #isExcluded = property(false);
+}
+
+export function isSplitTunnled(
+  response = {
+    t: "disabled_apps",
+    disabled_apps: [""],
+  }
+) {
+  if (response.t != "disabled_apps") {
+    throw new Error("passed an invalid response");
+  }
+  // Todo: THIS IS STILL HACKY
+  const search_terms = ["firefox.exe", "firefox"];
+  let apps = response.disabled_apps;
+  apps ??= [];
+  const isFirefoxExcluded = apps.some((path) => {
+    return search_terms.some((searchPath) => {
+      return path.endsWith(searchPath);
+    });
+  });
+  return isFirefoxExcluded;
```

Comment:
Just a thought: I wonder if we could verify this from the client? Like does the client have access to specific information about the Firefox instance it is communicating with and could that information be matched up against the excluded apps list?

Reply:
MAYBE - so the native messaging process is spawned by firefox, if they do not happen to use https://learn.microsoft.com/en-us/windows/win32/procthread/process-creation-flags?redirectedfrom=MSDN
We could in theory have bridge lookup it's parent process and find the path of the binary.